docs/troubleshooting.md

```diff
@@ -0,0 +1,29 @@
+# Troubleshooting Tips
+
+When developing operators, there might occasionally be moments when Kubernetes
+APIs or apiserver don't return error message with full context or when the error
+message can be a bit misleading. Purpose of this document is to list some common
+pitfalls that might be hard to debug in some cases and where returned error
+message is not perfectly clear.
+
+## RBAC rules
+
+When `informer.Watch()` is called, it might fail to unknown error:
+`unknown (get clusternetworkconfigs.core.giantswarm.io)`. This error is not very
```

Comment:
I would like to emphasize the importance of the error messages a bit more by structuring the paragraphs here in a way where we can support the reader visually a little bit more. I would like to have a separate block for the errors so your eyes can catch it fast. I would also like to have some more links to something like the status error or something like this. Being a bit detailed here helps a lot IMO. 

---

When [`informer.Watch()`](https://godoc.org/github.com/giantswarm/operatorkit/informer#Informer.Watch) is called, it might fail with an "unknown error". This error is not very descriptive and OperatorKit cannot really improve here since the original error is generic `StatusError`. Root cause here is a missing RBAC rule. The operator must have watch permission for the used CRD in order to be able to operate on it.

```
unknown (get clusternetworkconfigs.core.giantswarm.io)
```